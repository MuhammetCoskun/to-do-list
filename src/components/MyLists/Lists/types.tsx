import { ToDoListItemType } from "../../ShowList/ToDoList/types";

export type ListType = {
  id: string;
  title: string;
  todos: ToDoListItemType[];
};

export type ListsProps = {
  lists: ListType[];
  onChooseList: (arg: string) => void;
};
