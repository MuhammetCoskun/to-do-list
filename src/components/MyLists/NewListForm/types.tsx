export type NewListFormProps = {
  onAddList: (listName: string) => void;
};
