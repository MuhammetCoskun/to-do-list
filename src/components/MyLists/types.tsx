import { ListType } from "./Lists/types";

export type MyListsProps = {
  onChooseList: (arg: string) => void;
  onAddList: (title: string) => void;
  lists: ListType[];
};
