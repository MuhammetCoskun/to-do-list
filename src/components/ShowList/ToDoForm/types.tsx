export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
