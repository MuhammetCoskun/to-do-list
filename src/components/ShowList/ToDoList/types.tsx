import { ListType } from "../../MyLists/Lists/types";

export type ToDoListItemType = {
  id: string;
  text: string;
  createdAt: string;
  isDone: boolean;
  doneAt?: string;
};

export type ToDoListItemProps = {
  id: string;
  key?: string;
  text: string;
  onRemove: (id: string) => void;
};

export type ToDoListProps = {
  list: ListType;
  onRemove: (id: string) => void;
};
