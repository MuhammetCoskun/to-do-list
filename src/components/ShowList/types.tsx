import { ListType } from "../MyLists/Lists/types";

export type ShowListProps = {
  listToShow: ListType;
  onAddTodo: (text: string) => void;
  onRemoveTodo: (id: string) => void;
};
