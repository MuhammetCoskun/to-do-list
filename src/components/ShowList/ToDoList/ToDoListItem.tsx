import React from "react";
import { ToDoListItemProps } from "./types";

const ToDoListItem: React.FC<ToDoListItemProps> = ({
  id,
  text,
  isDone,
  onChangeIsDone,
}) => {
  const clickHandler = () => {
    onChangeIsDone(id);
  };

  return (
    <div className={"todo-task-item"}>
      <input type="checkbox" id={id} defaultChecked={isDone} />
      <label htmlFor={id} onClick={clickHandler}>
        <span className={"checkbox"}></span>
        {text}
      </label>
    </div>
  );
};

export default ToDoListItem;
