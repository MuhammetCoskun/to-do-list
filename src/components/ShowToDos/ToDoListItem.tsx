import React from "react";

const ToDoListItem: React.FC = () => {
  return <p>ToDoListItem Component</p>;
};

export default ToDoListItem;
