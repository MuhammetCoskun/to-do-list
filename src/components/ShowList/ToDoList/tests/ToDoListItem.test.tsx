import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ToDoListItem from "../ToDoListItem";
import { ToDoListItemProps } from "../types";

const customRender = (props: ToDoListItemProps) => {
  render(<ToDoListItem {...props} />);
};

const mockOnChangeIsDone = jest.fn();

describe("ToDoList Component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(
      <ToDoListItem
        id="id"
        text="text"
        onChangeIsDone={mockOnChangeIsDone}
        isDone={false}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("when todo item clicked it should call onChangeIsDone callback with id prop", () => {
    customRender({
      id: "id",
      text: "text",
      onChangeIsDone: mockOnChangeIsDone,
      isDone: false,
    });
    const todoItemElement = screen.getByTestId("label");
    userEvent.click(todoItemElement);
    expect(mockOnChangeIsDone).toHaveBeenCalledWith("id");
  });
});
