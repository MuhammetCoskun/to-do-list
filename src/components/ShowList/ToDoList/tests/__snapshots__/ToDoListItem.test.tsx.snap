// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="todo-task-item"
  >
    <input
      id="id"
      type="checkbox"
    />
    <label
      for="id"
    >
      <span
        class="checkbox"
      />
      text
    </label>
  </div>
</DocumentFragment>
`;
